semtypes/src/main/java/io/ballerina/semtype/Env.java

```diff
@@ -42,5 +42,17 @@ public Env() {
         this.recFunctionAtoms = new ArrayList<>();
     }
 
+    public synchronized RecAtom recFunctionAtom() {
+        int result = this.recFunctionAtoms.size();
+        this.recFunctionAtoms.add(null);
```

Comment:
shall we add a comment here saying we are adding a `()`

Reply:
Java is prone to get null pointer exceptions, so maybe `null` is not the best choice. On the other hand it is closer to the nballerina source.

Another option is to create a consent like below and use that instead. This is similar to what java Enums get desugared into.
```java
public static final FunctionAtomicType NIL_FUNCTION = new FunctionAtomicType();
```
@jclark any preference ?

Reply:
I'm going to merge this for now, @ushirask please create an issue if we get any change requests